File: src/Julia/Get-the-Middle-Character.jl
## Get-the-Middle-Character.jl

using Statistics


"""
return middle character(s) of a string
"""
function getmiddle(s::String)::String
  m = median(1:length(s))
  s[floor(Int, m):ceil(Int, m)] # median() 返回 Float
end


using Test
@testset "Sample tests" begin
  @test getmiddle("test") == "es"
  @test getmiddle("testing") == "t"
end

File: src/R/Get-the-Middle-Character.R
## Get the Middle Character

#' @title 提取字符串中间的一个或两个字母
get_middle <- function(string) {
  m <- median(1:nchar(string)) # middle letter index
  string |> str_sub(floor(m), ceiling(m))
}

library(testthat)
test_that("Sample Tests", {
  expect_equal(get_middle("test"), "es")
  expect_equal(get_middle("testing"), "t")
  expect_equal(get_middle("middle"), "dd")
  expect_equal(get_middle("A"), "A")
  expect_equal(get_middle("of"), "of")
})
